third_party/terraform/tests/resource_iap_tunnel_iam_test.go.erb

```diff
@@ -0,0 +1,295 @@
+			},
+			{
+				ResourceName:      "google_iap_tunnel_iam_binding.foo",
+				ImportStateId:     fmt.Sprintf("%s/%s/%s %s", project, zone, instanceName, role),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccIapTunnelIamMember(t *testing.T) {
+	t.Parallel()
+
+	project := getTestProjectFromEnv()
+	account := acctest.RandomWithPrefix("tf-test-sa")
+	role := "roles/iap.tunnelResourceAccessor"
+	region := getTestRegionFromEnv()
+	zone := getTestZoneFromEnv()
+	subnetwork := fmt.Sprintf("tf-test-net-%s", acctest.RandString(10))
+	instanceName := fmt.Sprintf("tf-test-instance-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				// Test Iam Member creation (no update for member, no need to test)
+				Config: testAccIapTunnelIamMember_basic(project, account, region, zone, subnetwork, instanceName, role),
+			},
+			{
+				ResourceName:      "google_iap_tunnel_iam_member.foo",
+				ImportStateId:     fmt.Sprintf("%s/%s/%s %s serviceAccount:%s@%s.iam.gserviceaccount.com", project, zone, instanceName, role, account, project),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccIapTunnelIamPolicy(t *testing.T) {
+	t.Parallel()
+
+	project := getTestProjectFromEnv()
+	account := acctest.RandomWithPrefix("tf-test-sa")
+	role := "roles/iap.tunnelResourceAccessor"
+	region := getTestRegionFromEnv()
+	zone := getTestZoneFromEnv()
+	subnetwork := fmt.Sprintf("tf-test-net-%s", acctest.RandString(10))
+	instanceName := fmt.Sprintf("tf-test-instance-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccIapTunnelIamPolicy_basic(project, account, region, zone, subnetwork, instanceName, role),
+			},
+			// Test a few import formats
+			{
+				ResourceName:      "google_iap_tunnel_iam_policy.foo",
+				ImportStateId:     fmt.Sprintf("projects/%s/zones/%s/instances/%s", project, zone, instanceName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				ResourceName:      "google_iap_tunnel_iam_policy.foo",
+				ImportStateId:     fmt.Sprintf("%s/%s/%s", project, zone, instanceName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				ResourceName:      "google_iap_tunnel_iam_policy.foo",
+				ImportStateId:     fmt.Sprintf("%s/%s", zone, instanceName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccIapTunnelIamBinding_basic(project, account, region, zone, subnetworkName, instanceName, roleId string) string {
+	return fmt.Sprintf(`
+resource "google_service_account" "test_account" {
+  account_id   = "%s"
+  display_name = "Iam Testing Account"
+}
+resource "google_compute_network" "network" {
```

Comment:
@rileykarson ack, I'm okay with switching to the default network, but how strongly do you feel about this? In my local testing, I use a long lived project where I've deleted the default network. This could cause issues in other people running the test if they do the same in their own projects.

Reply:
Not too strongly, but most of our tests rely on the default network already whether implicitly or explicitly so I'd say we can expect users to have one.

Reply:
thanks for the additional info. I can easily recreate my default network. good point on the network quota, and spinning up and down a network does take a long time!